src/CommandLineUtils/Conventions/OptionAttributeConventionBase.cs

```diff
@@ -57,6 +57,18 @@ private protected void AddOption(ConventionContext context, CommandOption option
                 context.Application._longOptions.Add(option.LongName, prop);
             }
 
+            if (option.OptionType != CommandOptionType.NoValue)
+            {
+                var getter = ReflectionHelper.GetPropertyGetter(prop);
+
+                var value = getter.Invoke(modelAccessor.GetModel());
```

Comment:
I started thinking about this attribute-binding use case a little more. I'm concerned this will change the lifecycle of initialization of the model class. This moves the order initialization into the execution of this convention. Depending on the order in which users have added to convention builder, the constructor injection may not be finalized yet. I didn't realize this before when I saw the first drafts of this PR, but the test change you made below to flip the order of convention calls made me realize that this is a problem.

Can we make this lazy? All other uses of `.GetModel()` in conventions are deferred using `OnValidate` or `OnParsingComplete`. Alternatively, we could introduce an abstraction like `IDefaultValueProvider<T>.GetDefault()` which could have one implementation for the builder API which binds to literal, statically defined values, and another one which has a callback to this reflection code.